connaisseur/alert.py

```diff
@@ -53,31 +54,66 @@ class AlertReceiverAuthentication:
     Class to store authentication information for securely sending events to the alert receiver.
     """
 
-    class AlertReceiverBasicAuthentication:
+    authentication_config: dict = None
+    authentication_scheme: str = None
+
+    class AlertReceiverAuthenticationInterface:
+        def __init__(self, alert_receiver_config: dict, authentication_key: str):
+            self.authentication_config = alert_receiver_config.get(authentication_key)
+
+            if self.authentication_config is None:
+                raise ConfigurationError(
+                    f"No authentication configuration found ({authentication_key})."
+                )
+
+            self.authentication_scheme = self.authentication_config.get(
+                "authentication_scheme", self.authentication_scheme
+            )
+            self._validate_authentication_scheme()
+
+        def _validate_authentication_scheme(self) -> None:
+            if not self.authentication_scheme:
+                raise ConfigurationError(
+                    "The authentication scheme cannot be null or empty."
+                )
+
+            if " " in self.authentication_scheme:
```

Comment:
You might want to check that it contains only a-zA-Z or something like that. (`" "` is very specific, doesn't catch tabs, newlines, ...)